storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp

```diff
@@ -992,6 +1008,92 @@ FileStorHandlerImpl::Stripe::getNextMessage(vespalib::steady_time deadline)
     return {}; // No message fetched.
 }
 
+FileStorHandler::LockedMessage
+FileStorHandlerImpl::Stripe::getNextMessage(vespalib::steady_time deadline)
+{
+    std::unique_lock guard(*_lock);
+    return next_message_impl(guard, deadline);
+}
+
+namespace {
+
+constexpr bool is_batchable_feed_op(api::MessageType::Id id) noexcept {
+    return (id == api::MessageType::PUT_ID ||
+            id == api::MessageType::REMOVE_ID ||
+            id == api::MessageType::UPDATE_ID);
+}
+
+// Precondition: msg must be a feed operation request (put, remove, update)
+document::GlobalId gid_from_feed_op(const api::StorageMessage& msg) {
+    switch (msg.getType().getId()) {
+    case api::MessageType::PUT_ID:
+        return dynamic_cast<const api::PutCommand&>(msg).getDocumentId().getGlobalId();
+    case api::MessageType::REMOVE_ID:
+        return dynamic_cast<const api::RemoveCommand&>(msg).getDocumentId().getGlobalId();
+    case api::MessageType::UPDATE_ID:
+        return dynamic_cast<const api::UpdateCommand&>(msg).getDocumentId().getGlobalId();
```

Comment:
dynamic_cast is a rather expensive operation. Consider replacing with static_cast or a getAsXXMsg cast safe method.

Reply:
Replaced with `static_cast`. Downcast-safe type invariant shall be maintained by the message's own type ID tracking. If it's not, we'd have other, bigger problems.